Sources/Moya/Response.swift

```diff
@@ -127,9 +127,18 @@ public extension Response {
             guard let jsonObject = (try mapJSON() as? NSDictionary)?.value(forKeyPath: keyPath) else {
                 throw MoyaError.jsonMapping(self)
             }
-            do {
-                jsonData = try JSONSerialization.data(withJSONObject: jsonObject)
-            } catch {
+            if let value = jsonObject as? D {
+                return value
+            }
+            let dict = jsonObject as? NSDictionary
```

Comment:
Instead of checking for `NSDictionary` or `NSArray`, can we use `isValidJSONObject` method? I feel like it will be more scalable this way. Also we can use `guard` to make the code below more user-friendly (with the check for valid JSON object)

Reply:
Yes, I'm pretty sure when I was digging through Foundation's source, I saw that `isValidJSONObject` just performs all the checks we're attempting here.

Reply:
Pretty much, yup: https://github.com/apple/swift-corelibs-foundation/blob/3ee0925fceb476e89b26673effb7b9f92edbc3da/Foundation/JSONSerialization.swift#L57-L126

Reply:
@BasThomas I seriously love how you're always willing to go grab a reference ❤️

Reply:
Let’s just say that
1. I was curious
2. I like context

Happy to help 😃